diff --git a/lib/chroot-setup.sh b/lib/chroot-setup.sh
index 4a92c4f..7663f3f 100644
--- a/lib/chroot-setup.sh
+++ b/lib/chroot-setup.sh
@@ -109,6 +109,15 @@ EOF
 	# Avoid apt-listchanges doing anything.
 	APT_LISTCHANGES_FRONTEND=none
 	export APT_LISTCHANGES_FRONTEND
+	cat > /target/etc/flash-kernel.conf <<FK
+#!/bin/sh
+
+echo "flash-kernel stopped:"
+echo "You are currently running the Community Kernel edition"
+echo "remove /etc/flash-kernel.conf to run Ubuntu's Kernel"
+exit
+
+FK
 
 	return 0
 }
