diff --git a/usr/lib/finish-install.d/08rcn-ee-finish-installing-device b/usr/lib/finish-install.d/08rcn-ee-finish-installing-device
new file mode 100755
index 0000000..6d06fc7
--- /dev/null
+++ b/usr/lib/finish-install.d/08rcn-ee-finish-installing-device
@@ -0,0 +1,32 @@
+#!/bin/sh -e
+cp /etc/e2fsck.conf /target/etc/e2fsck.conf
+cp /etc/finish-install.sh /target/etc/finish-install.sh
+chmod a+x /target/etc/finish-install.sh
+
+if [ -f /etc/rcn.conf ]; then
+	mkdir -p /target/boot/uboot
+	echo "/dev/mmcblk0p1   /boot/uboot auto   defaults            0   0" >> /target/etc/fstab
+
+	mount /dev/mmcblk0p1 /target/boot/uboot
+
+	#smsc95xx kevent workaround/hack
+	echo "vm.min_free_kbytes = 8192" >> /target/etc/sysctl.conf
+
+	if [ -d /lib/firmware/ ] ; then
+		cp -rf /lib/firmware/ /target/lib/ || true
+	fi
+
+	rm -f /etc/rcn.conf
+
+	mount -o bind /sys /target/sys
+	cat /proc/mounts > /target/mounts
+	cat /proc/mounts > /target/boot/uboot/backup/proc_mounts
+	chroot /target /bin/bash /etc/finish-install.sh
+	rm -f /target/mounts || true
+	cat /var/log/syslog > /target/boot/uboot/backup/syslog.log
+	umount /target/sys
+
+	sync
+	umount /target/boot/uboot
+fi
+
