diff --git a/usr/lib/finish-install.d/08rcn-omap b/usr/lib/finish-install.d/08rcn-omap
new file mode 100755
index 0000000..6d06fc7
--- /dev/null
+++ b/usr/lib/finish-install.d/08rcn-omap
@@ -0,0 +1,67 @@
+#!/bin/sh -e
+cp /etc/flash-kernel.conf /target/etc/flash-kernel.conf
+cp /etc/SERIAL.conf /target/etc/init/SERIAL.conf
+
+if [ -e /etc/rcn.conf ]; then
+	mkdir -p /tmp/mmc
+	mount /dev/mmcblk0p1 /tmp/mmc
+
+	#Remove NetInstall bits
+	rm -f  /tmp/mmc/uInitrd.net || true
+
+	#Cleanup Ubuntu's mess
+	rm -f /tmp/mmc/uInitrd || true
+	rm -f /tmp/mmc/uInitrd.bak || true
+	rm -f /tmp/mmc/uImage || true
+	rm -f /tmp/mmc/uImage.bak || true
+
+ if [ -e "/tmp/mmc/MLO.bak" ]; then
+  rm -f /tmp/mmc/MLO || true
+  mv /tmp/mmc/MLO.bak /tmp/mmc/MLO
+ fi
+
+ if [ -e "/tmp/mmc/u-boot.bin.bak" ]; then
+  rm -f /tmp/mmc/u-boot.bin || true
+  mv /tmp/mmc/u-boot.bin.bak /tmp/mmc/u-boot.bin
+ fi
+
+	#Install kernel image deb
+	LD_LIBRARY_PATH=/target/lib /target/usr/bin/dpkg-deb -x /tmp/mmc/linux-image-*_1.0*_armel.deb /target/
+	rm -f /tmp/mmc/linux-image-*_1.0*_armel.deb || true
+
+ #With Oneiric Updates, these now seem to ignore /etc/flash-kernel.conf, so lets just remove them..
+ chroot /target apt-get remove -y linux-image-omap* || true
+ chroot /target apt-get remove -y linux-headers-omap* || true
+ chroot /target apt-get remove -y u-boot-linaro* || true
+ chroot /target apt-get remove -y x-loader-omap* || true
+
+ chroot /target apt-get remove -y flash-kernel || true
+
+ #Setup Final Boot Scripts
+ if [ -e "/tmp/mmc/boot.scr" ]; then
+  rm -f /tmp/mmc/boot.scr || true
+  mv /tmp/mmc/normal.scr /tmp/mmc/boot.scr
+ fi
+
+	mkdir -p /target/boot/uboot
+	echo "/dev/mmcblk0p1   /boot/uboot auto   defaults            0   0" >> /target/etc/fstab
+
+	#smsc95xx kevent workaround/hack
+	echo "vm.min_free_kbytes = 8192" >> /target/etc/sysctl.conf
+
+	cp -r /lib/firmware/ /target/lib/ || true
+
+	rm -f /etc/rcn.conf
+
+	#Generate new initrd.img-*
+	mount -o bind /sys /target/sys
+	chroot /target update-initramfs -c -k `uname -r`
+	chroot /target mkimage -A arm -O linux -T ramdisk -C none -a 0 -e 0 -n initramfs -d /boot/initrd.img-`uname -r` /boot/uInitrd
+	cp /target/boot/uInitrd /tmp/mmc/uInitrd
+	chroot /target mkimage -A arm -O linux -T kernel -C none -a ZRELADD -e ZRELADD -n `uname -r` -d /boot/vmlinuz-`uname -r` /boot/uImage
+	cp /target/boot/uImage /tmp/mmc/uImage
+	umount /target/sys
+
+	sync
+	umount /tmp/mmc
+fi
+
